Disable local engines

r1jt opened the pull request:
This change will add a configuration variable to disable
 installation of local engines lvm and zfs during
 umbrella chart based installation

r1jt committed:
fix: add config to disable local engines

This change will add a configuration variable to disable
 installation of local engines lvm and zfs during
 umbrella chart based installation

Signed-off-by: Ranjith M P <92041707+r1jt@users.noreply.github.com>

blaisedias commented on common/e2e_config/e2e_config.go:
```diff
@@ -138,6 +138,7 @@ type ProductSpec struct {
 	LvmEnginePluginDriverName         string            `yaml:"lvmEnginePluginDriverName"`
 	UmbrellaOpenebsHelmChartName      string            `yaml:"umbrellaOpenebsHelmChartName"`
 	UseUmbrellaOpenEBSChart           bool              `yaml:"useUmbrellaOpenEBSChart" env:"e2e_use_umbrella_openebs_chart" env-default:"false"`
+	DisableLocalEngines           	  bool              `yaml:"disableLocalEngines" env:"e2e_disable_local_engines" env-default:"true"`
```
Please default the value to false, so that if environment variable is set local engines are installed - which should be the default behaviour.

r1jt replied:
Modified as suggested

rohan2794 commented on common/e2e_config/e2e_config.go:
```diff
@@ -138,6 +138,7 @@ type ProductSpec struct {
 	LvmEnginePluginDriverName         string            `yaml:"lvmEnginePluginDriverName"`
 	UmbrellaOpenebsHelmChartName      string            `yaml:"umbrellaOpenebsHelmChartName"`
 	UseUmbrellaOpenEBSChart           bool              `yaml:"useUmbrellaOpenEBSChart" env:"e2e_use_umbrella_openebs_chart" env-default:"false"`
+	DisableLocalEngines           	  bool              `yaml:"disableLocalEngines" env:"e2e_disable_local_engines" env-default:"false"`
```
don't use same flag to disable lvm and zfs  local engine, use separate flag for lvm and zfs

blaisedias replied:
Better still change to take a list of io-engines to disable - that scales better and reduces the number of flags

r1jt replied:
Why would we need separate flags? Do we need to test mayastor with either of the local engines installed?

rohan2794 replied:
local engines include lvm, zfs and hostpath but we only want to disable lvm and zfs here.

avishnu replied:
I think @blaisedias and @rohan2794 have a valid point. Have equivalent individual flags to turn on/off individual engines.

r1jt replied:
> Better still change to take a list of io-engines to disable - that scales better and reduces the number of flags
Added a flag which takes a list of io-engines

blaisedias commented:
Shouldn't  this feature be more generic? As in an option to disable any one of the engines?

r1jt commented:
> Shouldn't this feature be more generic? As in an option to disable any one of the engines?
Yes. I have now removed changes to handle environment variable setting

blaisedias commented:
> > Shouldn't this feature be more generic? As in an option to disable any one of the engines?
> > Yes. I have now removed changes to handle environment variable setting

The name is still `DisableLocalEngines` , it should be `DisableEngines`.
Configuration using an environment variable is a useful flexibility - it should be present.
The same means should be used to disable all IO engines including mayastor.

r1jt commented:
> > > Shouldn't this feature be more generic? As in an option to disable any one of the engines?
> > > Yes. I have now removed changes to handle environment variable setting
> 
> The name is still `DisableLocalEngines` , it should be `DisableEngines`. Configuration using an environment variable is a useful flexibility - it should be present. The same means should be used to disable all IO engines including mayastor.

I have modified the name to "DisableEngines". However environment variable is not present as I removed the logic to handle it in the pipelines code
